frontend/src/app/fullname.pipe.ts

```diff
@@ -0,0 +1,13 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { PersonModel } from './models/person.model';
+
+@Pipe({
+  name: 'fullname'
+})
+export class FullnamePipe implements PipeTransform {
+
+  transform(person: PersonModel): string {
+    const elements = [person.firstName, person.lastName, person.nickName ? `(${person.nickName})` : null];
+    return elements.filter(e => e).join(' ');
```

Comment:
The idea is to eliminate null values. Your code would print 'null null' if the first name and the last name are null.